### PR TITLE
sdk support code cleaup changes

### PR DESCRIPTION
**Pull request by sharad-patil24**
#### Problem / Feature
Matter support code cleanup

#### Change overview
    - Refactor Task management API's
    - Added missing function headers
    - Removed commented codes
    - Removed inconsistency between code and comment
    - Hardcoded and magic number removal.

#### Testing
How was this tested?
- Commissioning tested ( Multiple times )
- Sanity Testing done.
-

### CODE REVIEW

File: matter/wifi/wfx_notify.cpp [hunk 1]
```diff
@@ -34,90 +34,120 @@ using namespace ::chip::DeviceLayer;
...
 
-  if (status != 0) {
+  if (status != SUCCESS_STATUS) {
     EFR32_LOG("%s: error: failed status: %d.", __func__, status);
     return;
   }
 
   EFR32_LOG("%s: connected.", __func__);
 
-  memset(&evt, 0, sizeof evt);
```
Comment (jepenven-silabs) [inline matter/wifi/wfx_notify.cpp:74]: You can leave the 0 there.

Rule of the thumb for defines is to ask the following question : "Will I have to explain where this number comes from ?" if the answer is yes, add a define, if the answer is no don't add one. 

In this case you're clearly clearing a buffer, so no need to replace the `0` with a define

Comment (jepenven-silabs) [reply matter/wifi/wfx_notify.cpp:74]: Missing brackets `sizeof(evt)`

**Review by jepenven-silabs — CHANGES_REQUESTED** [review]
Good work! just a few small comments to address before merging !

**Review by jmartinez-silabs — APPROVED** [review]
Approved with the idea that JF's comments will be addresses and resolved

File: matter/wifi/ethernetif.cpp [hunk 2]
```diff
@@ -92,45 +97,51 @@ static void low_level_init(struct netif *netif)
...
+ * @param[in]  len: length
+ * @return
+ *     None
+ ************************************************************************************/
 static void low_level_input(struct netif *netif, uint8_t *b, uint16_t len)
 {
   struct pbuf *p, *q;
   uint32_t bufferoffset;
 
   if (len <= 0)
```
Comment (selissia) [inline matter/wifi/ethernetif.cpp:116]: It's a good practice to always use brackets {} in conditional statements.

File: matter/wifi/rs911x/hal/rsi_hal_mcu_rtc.c [hunk 3]
```diff
@@ -26,6 +26,13 @@
 #include <stdlib.h>
 #include <time.h>
 
+#define ZONE0 			0
+#define CH_SELECTOR 		1u
+#define CNT_VALUE 		0u
+#define RSI_RTC_FREQ_VALUE 	0
+#define TIME_ZONE_OFFSET 	0u
+#define VAL_1 			1
```
[the remark this answers is not in the document]
Comment (sharad-patil24) [reply matter/wifi/rs911x/hal/rsi_hal_mcu_rtc.c:34]: removed 2 macros 
CNT_VALUE & VAL_1 .

added comment subtraction of 1

File: matter/wifi/dhcp_client.cpp [hunk 4]
```diff
@@ -38,6 +38,7 @@
 #include "AppConfig.h"
 
 #define MAX_DHCP_TRIES 4
+#define NETIF_IPV4_ADDRESS(X) ((netif->ip_addr.u_addr.ip4.addr >> (8 * X)) && 0xFF)
```
[the remark this answers is not in the document]
Comment (sharad-patil24) [reply matter/wifi/dhcp_client.cpp:41]: done as per suggested Implementation from Jean.

File: matter/wifi/rs911x/hal/rsi_hal_mcu_rtc.c [hunk 5]
```diff
@@ -133,18 +140,18 @@ void rsi_rtc_init_timer(void) {
...
   // Compare channel starts disabled and is enabled only when compare match
   // interrupt is enabled.
   channel.chMode = rtccCapComChModeOff;
-  RTCC_ChannelInit(1u, &channel);
+  RTCC_ChannelInit(CH_SELECTOR, &channel);
 
   RTCC_IntDisable(_RTCC_IEN_MASK);
   RTCC_IntClear(_RTCC_IF_MASK);
-  RTCC_CounterSet(0u);
+  RTCC_CounterSet(CNT_VALUE);
```
[the remark this answers is not in the document]
Comment (sharad-patil24) [reply matter/wifi/rs911x/hal/rsi_hal_mcu_rtc.c:154]: Renamed to RTC_DEFAULT_COUNTER_VALUE

File: matter/wifi/wfx_notify.cpp (hunk 1 above)
Comment (sharad-patil24) [reply matter/wifi/wfx_notify.cpp:74]: removed CLEAR_BUFFER where it is used  and added brackets around sizeof

File: matter/wifi/ethernetif.cpp (hunk 2 above)
Comment (sharad-patil24) [reply matter/wifi/ethernetif.cpp]: done

**Review by jmartinez-silabs — APPROVED** [review]

**Pull request merged**